File: frontend/lib/diet/models/dietary_preference.dart
/// Dietary preference categories (backend `DietaryPreference`).
enum DietaryPreference {
  vegetarian('VEGETARIAN'),
  vegan('VEGAN'),
  nonVegetarian('NON_VEGETARIAN'),
  pescatarian('PESCATARIAN'),
  other('OTHER');

  const DietaryPreference(this.wireName);

  /// The exact value sent to / received from the backend.
  final String wireName;

  String get label {
    return switch (this) {
      DietaryPreference.vegetarian => 'Vegetarian',
      DietaryPreference.vegan => 'Vegan',
      DietaryPreference.nonVegetarian => 'Non-vegetarian',
      DietaryPreference.pescatarian => 'Pescatarian',
      DietaryPreference.other => 'Other',
    };
  }

  static DietaryPreference fromWire(String? value) {
    return DietaryPreference.values.firstWhere(
      (preference) => preference.wireName == value,
      orElse: () => DietaryPreference.other,
    );
  }
}

File: backend/src/main/java/com/blistra/diet/domain/DietaryPreference.java
package com.blistra.diet.domain;

/**
 * Dietary preference categories.
 *
 * <p>The taxonomy is intentionally small. {@code OTHER} allows a user-supplied
 * label (validated, length-limited) instead of a huge closed taxonomy.</p>
 */
public enum DietaryPreference {
    VEGETARIAN,
    VEGAN,
    NON_VEGETARIAN,
    PESCATARIAN,
    OTHER
}